{
  "error": {
    "message1": "error[E0999]: assignment might be unsafe\n   --> src/vec_deque.rs:512:9\n    |\n512 |         self.tail = self.wrap_sub(tail, 1);\n    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\n",
    "message2": "error[E0999]: failed to verify predicate: \"a4 = a3.1\"\n              blamed variable: \"Some(a4)\"\n              related variables: [{\"name\": \"a4\", \"depth\": 4, \"origin\": \"(CallReturn, Some(src/vec_deque.rs:512:21: 512:43 (#0)))\"}, {\"name\": \"a3\", \"depth\": 3, \"origin\": \"No provenance\"}]\n   --> src/vec_deque.rs:512:9\n    |\n512 |         self.tail = self.wrap_sub(tail, 1);\n    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n    |\nnote: related variable `a4 (depth 4)` defined here with originator `CallReturn`\n   --> src/vec_deque.rs:512:21\n    |\n512 |         self.tail = self.wrap_sub(tail, 1);\n    |                     ^^^^^^^^^^^^^^^^^^^^^^\n\n",
    "file": "/home/cole/research/flux-examples/kani-vecdeque/src/vec_deque.rs",
    "row": 512,
    "col": 9,
    "raw1": "error[E0999]: assignment might be unsafe\n   --> src/vec_deque.rs:512:9\n    |\n512 |         self.tail = self.wrap_sub(tail, 1);\n    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\n",
    "raw2": "error[E0999]: failed to verify predicate: \"a4 = a3.1\"\n              blamed variable: \"Some(a4)\"\n              related variables: [{\"name\": \"a4\", \"depth\": 4, \"origin\": \"(CallReturn, Some(src/vec_deque.rs:512:21: 512:43 (#0)))\"}, {\"name\": \"a3\", \"depth\": 3, \"origin\": \"No provenance\"}]\n   --> src/vec_deque.rs:512:9\n    |\n512 |         self.tail = self.wrap_sub(tail, 1);\n    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n    |\nnote: related variable `a4 (depth 4)` defined here with originator `CallReturn`\n   --> src/vec_deque.rs:512:21\n    |\n512 |         self.tail = self.wrap_sub(tail, 1);\n    |                     ^^^^^^^^^^^^^^^^^^^^^^\n\n",
    "predicate": "a4 = a3.1",
    "blamed_variable": "Some(a4)",
    "related_variables": [
      {
        "name": "a4",
        "depth": 4,
        "origin": "(CallReturn, Some(src/vec_deque.rs:512:21: 512:43 (#0)))"
      },
      {
        "name": "a3",
        "depth": 3,
        "origin": "No provenance"
      }
    ],
    "parsing_error": null,
    "function_context": {
      "start": 505,
      "end": 517,
      "name": "push_front",
      "constraint_files": [],
      "rendered_context": "\nFunction context (/home/cole/research/flux-examples/kani-vecdeque/src/vec_deque.rs:505-517):\nFunction name: push_front\n---------------------------------------------------------------------------------------------\n 505:            //#[stable(feature = \"rust1\", since = \"1.0.0\")]\n 506:            pub fn push_front(&mut self, value: T) {\n 507:                if self.is_full() {\n 508:                    self.grow();\n 509:                }\n 510:        \n 511:                let tail = self.tail;\n 512: error>         self.tail = self.wrap_sub(tail, 1);\n 513:                let tail = self.tail;\n 514:                unsafe {\n 515:                    self.buffer_write(tail, value);\n 516:                }\n 517:            }\n"
    },
    "unique_name": "push_front-L512-0"
  },
  "fix": {
    "certainty": true,
    "err_type": "2a",
    "fix_description": {
      "fix_type": "additional_refinement",
      "description": "wrap_sub unrefined (needs to relate to tail)"
    },
    "seen_before": false,
    "blamed_variable_correct": true,
    "variables_to_change": [
      {
        "name": "a4",
        "depth": 4,
        "origin": "(CallReturn, Some(src/vec_deque.rs:512:21: 512:43 (#0)))"
      },
      {
        "name": "a3",
        "depth": 3,
        "origin": "No provenance"
      }
    ],
    "anything_missing": false,
    "missing_info": null,
    "additional_notes": null
  }
}